app/src/main/java/app/musikus/ui/activesession/ActiveSessionUiState.kt

```diff
@@ -8,20 +8,144 @@
+)
+
+@Stable
+/** Main UI State */
 data class ActiveSessionUiState(
+    val topBarUiState: StateFlow<ActiveSessionTopBarUiState> = MutableStateFlow(ActiveSessionTopBarUiState()),
+    val mainContentUiState: StateFlow<MainContentUiState> = MutableStateFlow(MainContentUiState()),
+    val newItemSelectorUiState: StateFlow<NewItemSelectorUiState> = MutableStateFlow(NewItemSelectorUiState()),
+    val toolsUiState: StateFlow<ActiveSessionToolsUiState> = MutableStateFlow(ActiveSessionToolsUiState()),
+    val dialogVisibilities: StateFlow<DialogVisibilities> = MutableStateFlow(DialogVisibilities()),
+)
+
+data class DialogVisibilities(
+    val newItemSelectorVisible: Boolean = false,
+    val finishDialogVisible: Boolean = false,
+    val discardDialogVisible: Boolean = false,
+    val createItemDialogVisible: Boolean = false,
+    val createFolderDialogVisible: Boolean = false,
+)
+
+@Stable
+data class MainContentUiState(
+    val timerUiState: StateFlow<ActiveSessionTimerUiState> = MutableStateFlow(ActiveSessionTimerUiState()),
+    val currentItemUiState: StateFlow<ActiveSessionCurrentItemUiState> = MutableStateFlow(ActiveSessionCurrentItemUiState()),
+    val pastSectionsUiState: StateFlow<ActiveSessionCompletedSectionsUiState> = MutableStateFlow(ActiveSessionCompletedSectionsUiState()),
+    val endDialogUiState: StateFlow<ActiveSessionEndDialogUiState> = MutableStateFlow(ActiveSessionEndDialogUiState())
+)
+
+@Stable
+data class ActiveSessionTopBarUiState(
+    val visible: Boolean = false,
+    val pauseButtonAppearance: SessionPausedResumedState = SessionPausedResumedState.RUNNING,
+)
+
+@Stable
+data class ActiveSessionTimerUiState(
+    val timerText: String = "00:00", // Prevents size change animation to trigger when opening the screen
+    val subHeadingAppearance: SessionPausedResumedState = SessionPausedResumedState.RUNNING,
+    val subHeadingText: String = ""
+)
+
+@Stable
+data class ActiveSessionCurrentItemUiState(
+    val visible: Boolean = false,
+    val name: String = "",
+    val color: Color = Color.Transparent,
+    val durationText: String = "",
+)
+
+@Stable
+data class ActiveSessionCompletedSectionsUiState(
+    val visible: Boolean = false,
+    val items: List<CompletedSectionUiState> = emptyList()
+)
+
+@Stable
+data class ActiveSessionEndDialogUiState(
+    val rating: Int = 5,
+    val comment: String = "",
+)
+
+@Stable
+data class CompletedSectionUiState(
+    val id: UUID = UUIDConverter.deadBeef,
+    val name: String = "",
+    val durationText: String = "",
+    val color: Color = Color.Transparent,
+)
+
+@Stable
+data class NewItemSelectorUiState(
+    val runningItem: LibraryItem? = null,
+    val foldersWithItems: List<LibraryFolderWithItems> = emptyList(),
+    val lastPracticedDates: Map<UUID, ZonedDateTime?> = emptyMap(),
+    val rootItems: List<LibraryItem> = emptyList()
+)
+
+@Stable
+data class ActiveSessionToolsUiState(
+    val activeTab: ActiveSessionTab = ActiveSessionTab.entries[0],
```

Comment:
I like to add a companion object to classes like `ActiveSessionTab` which includes a constant value `DEFAULT`